lesson12/js/main.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+const menuContainer = document.getElementById('js-menu_container');
+const menuLists = document.createElement('ul');
+
+function showLoadingImg() {
+	const createLoadingBox = document.createElement('div');
+	createLoadingBox.classList.add('ly_loading');
+	createLoadingBox.id = 'js-loading';
+	const loadingImg = document.createElement('img');
+	loadingImg.src = 'images/loading-circle.gif';
+	loadingImg.classList.add('el_loadingImg');
+	const getBody = document.getElementsByTagName('body');
+	getBody[0].appendChild(createLoadingBox).appendChild(loadingImg);
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function createMenuList(data) {
+	const fragment = document.createDocumentFragment();
+	menuLists.innerHTML = '';
```

Comment:
When I click the button, the menu grows infinitely, so I tried to initialize the list. Is this the method?

Reply:
ok,
improve your code

```js
function removeInnerHTML(element){
 element.innerHTML = "";
}

function createMenuList(data) {
	const fragment = document.createDocumentFragment();
	removeInnerHTML(menuLists) // 何度もボタン押下した際に積まれていくので都度初期化している

Reply:
other solve
you can pass third argument for addEventListener as option. once,
User can click the button only once.
add `disabled true` for button style.
```js
clickBtn.addEventListener(
  "click",
  () => {
    callMenuContents();
  },
  { once: true }
);
```

Reply:
Thank you for review!

I implemented only one click received in the review with the following PR.
[once Click ver](https://github.com/nyattsu72/hanson-frontend-cource/pull/16)
May I ask you to confirm?